### ISSUE TITLE
`plot_unit_waveforms` and `plot_unit_templates` display incorrectly for probegroup

### ISSUE THREAD

**Issue by zm711** [SpikeInterface/spikeinterface#2222]
Here is one dataset:

![image](https://github.com/SpikeInterface/spikeinterface/assets/92116279/94ec4895-4244-40ec-8cdb-467114600cbc)

Here's another dataset:

![image](https://github.com/SpikeInterface/spikeinterface/assets/92116279/627c9d54-f521-4aaf-95ac-3975cad00b94)

The difference between the datasets are that the second has a probegroup attached rather than a probe. But other than that they should be equivalent. 

In fact the x distance for the probe is 25 um and for the probegroup, the distance on that shank is only 20 um so this isn't a distance thing. I've tried setting the figsize myself and that didn't help.

This is currently in the `matplotlib` backend.

**Comment by alejoe91** [comment]
Interesting and weird! @zm711 can you show us how you are attaching the probe/probegroup?

**Comment by zm711** [comment]
This the probegroup we made on the probe interface issue together. and it is just a 

```python
recording.set_probegroup(multishank)
```

for the weird looking one.

**Comment by alejoe91** [comment]
and the other one is `recording.set_probe(multishank)`?

Can you share the probeinterface JSON?
```
pi.write_probeinterface("multishank.json", multishank)
```

**Comment by zm711** [comment]
> and the other one is recording.set_probe(multishank)

No. So it is a different probe. It's just one of the commercial probes.

so,
```python
recording.set_probe(commercial_probe)
```

Yep. It's on my work computer, so I'll send it later today (both the probe and the probe group).

**Comment by zm711** [comment]
[multishank.json](https://github.com/SpikeInterface/spikeinterface/files/13415379/multishank.json)
[singleprobe.json](https://github.com/SpikeInterface/spikeinterface/files/13415380/singleprobe.json)
@alejoe91, here are two json files. The multishank was a probegroup and the singleprobe was set with a `set_probe`.

**Comment by zm711** [comment]
I figured it out:

So the probe I was using is not aligned in the x (ie it makes a v), so the `get_waveforms_scales` function fails to get the correct x offset. I'm not sure of the exact correct solution but I'll link a PR that can at least start a discussion.
To get this corrected image I just found that if the `delta_x` is too small (ie in my case of the slowly v electrodes the delta x was .5 which is basically 0 and was making too small an offset.
![image](https://github.com/SpikeInterface/spikeinterface/assets/92116279/6165f3a3-5a93-47dc-a6ef-2868927640ae)
